docs/src/tutorial.md

```diff
@@ -19,54 +19,52 @@ end
 # define a dense layer with input size i and output size o
 Dense(i :: Int, o :: Int, f=sigm) = Dense(param(o, i), param0(o), f)
 ```
-More layer types can be defined.
+More sophisticated layers may be defined.
 Once again, see the [Knet.jl tutorial](https://github.com/denizyuret/Knet.jl/tree/master/tutorial) for more details.
 
-### Definition of the chained structure that evaluates the network and the loss function (negative log likelihood)
+### Definition of the loss function (negative log likelihood)
+Next, we define a chain that stacks layer, to evaluate them successively.
 ```@example KnetNLPModel
 using KnetNLPModels
 
-struct ChainNLL <: Chain # must derive from KnetNLPModels.Chain
+struct ChainNLL
   layers
   ChainNLL(layers...) = new(layers)
 end
 (c :: ChainNLL)(x) = (for l in c.layers; x = l(x); end; x) # evaluate the network for a given input x
 (c :: ChainNLL)(x, y) = Knet.nll(c(x), y) # compute the loss function given input x and expected output y
 (c :: ChainNLL)(data :: Tuple{T1,T2}) where {T1,T2} = c(first(data,2)...) # evaluate loss given data inputs (x,y)
 (c :: ChainNLL)(d :: Knet.Data) = Knet.nll(c; data=d, average=true) # evaluate loss using a minibatch iterator d
+
+DenseNet = ChainNLL(Dense(784, 200), Dense(200, 50), Dense(50, 10))
 ```
-The chained structure that defines the neural network must be a subtype of `KnetNLPModels.Chain`.
 
 ### Load datasets and define minibatch
-In this example, we use the [MNIST](https://juliaml.github.io/MLDatasets.jl/stable/datasets/MNIST/) dataset from [MLDatasets.jl](https://github.com/JuliaML/MLDatasets.jl.git).
+In this example, and to fit the architecture proposed, we use the [MNIST](https://juliaml.github.io/MLDatasets.jl/stable/datasets/MNIST/) dataset from [MLDatasets.jl](https://github.com/JuliaML/MLDatasets.jl.git).
 ```@example KnetNLPModel
 using MLDatasets
 
 # download datasets without user intervention
 ENV["DATADEPS_ALWAYS_ACCEPT"] = true 
 
-xtrn, ytrn = MNIST.traindata(Float32) # MNIST training dataset
-ytrn[ytrn.==0] .= 10 # re-arrange indices
-xtst, ytst = MNIST.testdata(Float32) # MNIST test dataset
-ytst[ytst.==0] .= 10 # re-arrange indices
+T = Float32
+xtrain, ytrain = MLDatasets.MNIST(Tx = T, split = :train)[:] 
+xtest, ytest = MLDatasets.MNIST(Tx = T, split = :test)[:] 
+ytrain[ytrain.==0] .= 10 # re-arrange indices
+xtest, ytest = MNIST.testdata(Float32) # MNIST test dataset
+ytest[ytest.==0] .= 10 # re-arrange indices
 
-dtrn = minibatch(xtrn, ytrn, 100; xsize=(size(xtrn, 1), size(xtrn, 2), 1, :)) # training minibatch
-dtst = minibatch(xtst, ytst, 100; xsize=(size(xtst, 1), size(xtst, 2), 1, :)) # test minibatch
+dtrn = minibatch(xtrain, ytrain, 100; xsize=(size(xtrain, 1), size(xtrain, 2), 1, :)) # training minibatch
```

Comment:
would it be better to create a variable call b_size and pass it so the reader understand the batch size is 100

Reply:
We should clarify (homogenize) the minibatch sizes.
Because, in this case, 100 indicates that each minibtach is a hundreth (1/100) of the given dataset.
For example, a MNIST minibatch is of size 600 (60000/100)